### PR TITLE
Add `OneCycle` constructor

### LINKED ISSUE #42: OneCycle annealing
### Describe the potential feature

One cycle annealing (original paper [here](https://arxiv.org/abs/1708.07120)) is a really strong scheduler and what I've found to be the most optimal scheduler for deep learning. An implementation in PyTorch is here: https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.OneCycleLR.html

### Motivation

_No response_

### Possible Implementation

```julia
"Cosine anneal from `start` to `end` as pct goes from 0.0 to 1.0."
function annealing_cos(start, stop, pct)
    cos_out = cos(pi * pct) + 1
    return stop + (start - stop) / 2.0 * cos_out
end
```

You would typically start at pct 0.3, and go to pct=1.0 at the end of training.

### PR DESCRIPTION
**Pull request by darsnack**
Closes #42 by adding a `OneCycle` convenience constructor.

### PR Checklist

- [x] Tests are added
- [x] Documentation, if applicable

### CODE REVIEW

**Comment by isentropic** [comment]
anything i can do to get this one through? I kidna need the onecycle and it'd be good to make available for everyone?
also the docs are not plotting exp scehdule here https://fluxml.ai/ParameterSchedulers.jl/dev/

**Comment by darsnack** [comment]
Nightly failures are unrelated (due to Zygote compilation).

**Review by ToucheSir — APPROVED** [review]
